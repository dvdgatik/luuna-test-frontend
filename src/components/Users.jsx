import React from 'react';
import styled from 'styled-components';
const DivCard = styled.div`
  flex: 1 0 30% !important;
  border-radius: 20px !important;
  box-shadow: 1px 2px 3px #c6c6c6;
  margin: 20px !important;

`;

const Img  = styled.img`
  border-radius:50% !important;
  width: 30%;
  display:block;
  margin-left:auto;
  margin-right:auto;
  margin-top:20px;
  margin-bottom:20px;
`;

const H3 = styled.h3`
  text-align:center;
  color:#F71111;
`;

const Users = props => {

	return(
    <>
    <H3>{props.notfound}</H3>
		<div className="card-group">
    {

      props.users.map((value, key)=>(
      <DivCard key={key} className="card">
        <div className="card-body ">
          <h5 className="card-title"><strong>Username: </strong> <a href={value.html_url}>{value.login}</a></h5>
          <p className="card-text">Followers</p>
          <p className="card-text text-center">
          <Img className="card-img-top" src={value.avatar_url} alt="Card image cap"/>
          </p>
        </div>
       <div className="card-footer">
          <small className="text-muted"><strong>
          <a href={value.html_url}>{value.html_url}</a>
          </strong></small>
        </div>
      </DivCard>
        ))
    }
  </div>
  </>
	)
}

export default Users;