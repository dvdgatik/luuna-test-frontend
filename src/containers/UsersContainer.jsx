import React, {useState, useEffect} from 'react';
import Users from '../components/Users';
import getData from '../utils/getData.js';
import Search from '../components/Search';
import Title from '../components/Title';
const API = 'https://api.github.com/users';

const UsersContainer = props => {
	const [users, setUsers] = useState([]);
	const [userinput, setUserInput] =  useState('');
	const [notfound, setNotFound] = useState('');
	const handleSearch = event => {
		event.persist();
		setUserInput(event.target.value);
		//console.log(userinput);
	}
	const handleClick = (e) => {
		event.preventDefault();
		getData(API+'/'+userinput)
		.then(res => {
			//console.log([res.data]);
			setUsers([res.data]);
			setNotFound('');
			//console.log(users);
		})
		.catch(err => {
			setNotFound('User Not Found');
			//console.log(notfound);
		});
	}
	useEffect(()=>{
		let isMounted = true;

		getData(API)
		.then(response => {
			if(isMounted) {
				setUsers(response.data);
			}
		})
		.catch(err=>err);

		return () => {
      		// clean up
      		isMounted = false;
    	};


	}, [userinput]);



	



	return(
		<>
		<Title title={'Users'}/>
		<Search  handleClick={handleClick} handleSearch={handleSearch} name_input={'Users...'}/>
		<Users notfound={notfound} users={users}/>
		</>
	)
}

export default UsersContainer;