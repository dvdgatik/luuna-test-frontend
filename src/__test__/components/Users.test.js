import React from 'react';
import {mount, shallow} from 'enzyme';
import UsersContainer from '../../containers/UsersContainer';
import Users from '../../components/Users';
import {create} from 'react-test-renderer';


describe('<UsersContainer/>',()=>{
	const users_cont = shallow(<UsersContainer/>);
	test('Render of UsersContainer Component',()=>{
		expect(users_cont.length).toEqual(1);
	});

	const users = shallow(<Users/>);
	test('Render of Users Component', ()=>{
		expect(users.length).toEqual(1);
	});


});



describe('Users Snapshot', () => {
	test('Check UI of Users Component',()=>{
		const users = create(<Users/>);
		expect(users.toJSON()).toMatchSnapshot();
	})
});