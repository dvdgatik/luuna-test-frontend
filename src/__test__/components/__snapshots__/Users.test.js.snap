// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Snapshot Check UI of Users Component 1`] = `
Array [
  <h3
    className="sc-dlnjwi ikjnRt"
  />,
  <div
    className="card-group"
  />,
]
`;
